### PR TITLE
Fix `unified_log` handling of timestamp formats

### PR DESCRIPTION
**Pull request by Smjert**
The log entries timestamp is a double, we need to ensure that it's cast to an integer when returning it as a result to sqlite and using a timestamp constraint, otherwise sqlite will filter out all the entries that have the decimal part in them.

### CODE REVIEW

File: osquery/tables/system/darwin/unified_log.mm [hunk 1]
```diff
@@ -380,7 +380,8 @@ QueryData genUnifiedLog(QueryContext& queryContext) {
 
         Row r;
 
-        r["timestamp"] = BIGINT([[entry date] timeIntervalSince1970]);
+        r["timestamp"] = BIGINT(
+            static_cast<std::uint32_t>([[entry date] timeIntervalSince1970]));
```
Comment (directionless) [inline osquery/tables/system/darwin/unified_log.mm:384]: In the spirit of not losing information, do you think it makes sense to also pass a float through? (`timestamp_float` or something)

Comment (Smjert) [reply osquery/tables/system/darwin/unified_log.mm:384]: Sure (I would call it `timestamp_double` to respect the true type).

**Comment by Smjert** [comment]
There's an additional problem I did not consider. When writing a constraint like `timestamp > 123`, the predicate passed to the unified log is actually `timestamp 123.0` which then means that entries with timestamp `123.1`, `123.2` and so on are for all intents and purposes higher than that, but then when casted back to an integer they are still 123.
So these will still get filtered by sqlite and at the same time count towards the default 100 rows returned limit.

The solution here is to actually pass the timestamp in the constraint + 1. If it was any other logging system, a log message with time `123.1` would've had `123` as a timestamp.
The table has `predicate` that can be used to do more precise constraints in theory.

**Comment by Smjert** [comment]
This should do, although there are still some interesting questions around the `>` and `>=` priorities on the `timestamp` column due to how they are processed (where `>=` is "stronger" if the constraint value is the same between `>` and `>=`, but whoever has the value that's higher, wins). This can wait though.

**Review by directionless — APPROVED** [review]

**Pull request merged**